cdap-security/src/main/java/co/cask/cdap/security/store/FileSecureStore.java

```diff
@@ -102,39 +99,26 @@
    * @param name Name of the element to store.
    * @param data The data that needs to be securely stored.
    * @param description User provided description of the entry.
-   * @param properties Metadata associated with the data  @throws IOException
+   * @param properties Metadata associated with the data
+   * @throws IOException If it failed to store the key in the store.
    */
   @Override
   public void put(String namespace, String name, byte[] data, String description, Map<String, String> properties)
     throws IOException {
     String keyName = getKeyName(namespace, name);
     SecureStoreMetadata meta = SecureStoreMetadata.of(name, description, properties);
     SecureStoreData secureStoreData = new SecureStoreData(meta, data);
-    Key key = null;
     writeLock.lock();
     try {
       if (keyStore.containsAlias(keyName)) {
-        key = keyStore.getKey(keyName, password);
+        throw new IOException("The key " + name + " under namespace " + namespace + "already exists.");
```

Comment:
should create a class called ``SecureKey`` that extends ``EntityId`` and implements ``NamespacedId`` in ``cdap-proto``, and throw a NotFoundException with an object of that class here. The REST APIs will then respond with the right code automatically